test/replaytests/replay_test.go

```diff
@@ -99,6 +99,33 @@ func TestGreetingsWorkflow2(t *testing.T) {
 	require.Error(t, err)
 }
 
+// This test registers Cherry Activity as the activity and executes the same activity in the workflow.
+// Ideally replayer doesn't concern itself with the change in the activity content until it matches the expected output type.
+// History has recorded the output of banana activity instead. Here, The workflow is not waiting for the activity so it doesn't notice
+// that activity id different from the one recorded in history.
+func TestExclusiveChoiceWorkflowWithUnregisteredActivity(t *testing.T) {
+	replayer := worker.NewWorkflowReplayer()
+
+	replayer.RegisterWorkflowWithOptions(exclusiveChoiceWorkflow, workflow.RegisterOptions{Name: "choice"})
+	replayer.RegisterActivityWithOptions(orderCherryActivity, activity.RegisterOptions{Name: "testactivity", DisableAlreadyRegisteredCheck: true})
+	err := replayer.ReplayWorkflowHistoryFromJSONFile(zaptest.NewLogger(t), "choice.json")
+	require.NoError(t, err)
+}
```

Comment:
getOrderActivity is the default activity that was recorded in the History so we don't need to register is separately I think.
Yes, the recorded history is banana but we are registering cherry.

Reply:
> Yes, the recorded history is banana but we are registering cherry.

and

> and executes the same activity in the workflow.

disagree, because it's executing banana.

---

re registering separately: given other tests / etc, I think we kinda have two options for how we think replayer should work, and the tests are currently claiming inconsistent things:
1. we should make sure activities are registered, and that the workflow calls the same activities.
    - in this case, the `getOrderActivity` call should fail.  currently it succeeds.
2. we shouldn't care, because it doesn't actually matter for _replaying a workflow_.  the data can still be read, because it has nothing to do with the activity, only the `.Get(ctx, &out)` out-param's type.
    - in this case, `TestGreetingsWorkflowforActivity` should be fine because it's not registered, but currently it fails.

Reply:
I think we agree that it should be 1, but in that case these tests should be "correct" up to the point that they're not.  otherwise there's N things that are wrong and they're claiming it's Z when it could / should be X or Y which happened earlier.

Reply:
(I think some of this is also that I'm getting confused by the blend of "test passes but it should not" when I'm rereading, so I'll double check all this later with a fresh brain and see if I'm missing something.)